File: app/src/main/java/app/rpgbuilder/adapters/options/OptionsRepository.java
package app.rpgbuilder.adapters.options;

import app.rpgbuilder.R;

public final class OptionsRepository {
    private OptionsRepository() {}

    private static Option[] options = new Option[] {
        new Option("Adventure", "Test", R.drawable.sword),
            new Option("Library", "Test", R.drawable.book),
            new Option("Jukebox", "Test", R.drawable.sword),
            new Option("Options", "Test", R.drawable.book),
            new Option("Builder", "Test", R.drawable.book)
    };

    public static int size() {
        return options.length;
    }

    public static Option getOption(int index) {
        return options[index];
    }
}

File: app/src/main/java/app/rpgbuilder/tmpkt/activities/OptionsRepository.kt
package app.rpgbuilder.tmpkt.activities

import app.rpgbuilder.R

class OptionsRepository {

    class Option constructor(
        internal val name: String,
        internal val description: String,
        internal val imageResource: Int,
        internal val newer: Boolean = false
    )

    companion object {
        @JvmStatic
        private val options = arrayOf(
            Option("Adventure", "Test", R.drawable.sword),
            Option("Library", "Test", R.drawable.book),
            Option("Jukebox", "Test", R.drawable.sword),
            Option("Options", "Test", R.drawable.book),
            Option("Builder", "Test", R.drawable.book)
        )

        @JvmStatic
        fun size(): Int = options.size

        @JvmStatic
        fun getOption(position: Int): Option = options[position]
    }
}